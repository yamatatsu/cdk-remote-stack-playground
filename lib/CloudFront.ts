import * as cloudfront from "@aws-cdk/aws-cloudfront";
import * as cdk from "@aws-cdk/core";

type Props = cdk.StackProps & {
  webAclArn: string;
};

export class CloudFrontStack extends cdk.Stack {
  constructor(scope: cdk.App, id: string, props: Props) {
    super(scope, id, props);

    new cloudfront.CloudFrontWebDistribution(this, "Distribution", {
      webACLId: props.webAclArn,
      originConfigs: [
        {
          customOriginSource: {
            domainName: "dev.classmethod.jp",
          },
          behaviors: [
            {
              isDefaultBehavior: true,
            },
          ],
        },
      ],
    });
  }
}
