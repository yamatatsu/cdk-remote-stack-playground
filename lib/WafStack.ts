import * as wafv2 from "@aws-cdk/aws-wafv2";
import * as cdk from "@aws-cdk/core";

export class WebAclStack extends cdk.Stack {
  public readonly webAclArn: string;

  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const acl = new wafv2.CfnWebACL(this, "CommonAcl", {
      defaultAction: {
        allow: {},
      },
      description: "CommonAcl",
      name: "CommonAcl",
      rules: [
        {
          name: "AWSCommonRule",
          priority: 0,
          overrideAction: {
            none: {},
          },
          visibilityConfig: {
            sampledRequestsEnabled: true,
            cloudWatchMetricsEnabled: true,
            metricName: "RuleWithAWSManagedRulesMetric",
          },
          statement: {
            managedRuleGroupStatement: {
              excludedRules: [],
              name: "AWSManagedRulesCommonRuleSet",
              vendorName: "AWS",
            },
          },
        },
      ],
      scope: "CLOUDFRONT",
      visibilityConfig: {
        sampledRequestsEnabled: true,
        cloudWatchMetricsEnabled: true,
        metricName: "CommonAclMetric",
      },
    });

    this.webAclArn = acl.attrArn;
  }
}
