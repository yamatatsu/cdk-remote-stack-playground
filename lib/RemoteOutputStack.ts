import { RemoteOutputs } from "cdk-remote-stack";
import * as cdk from "@aws-cdk/core";
import { WebAclStack } from "./WafStack";

const outputKey = "WebAclArn";

type Props = cdk.StackProps & {
  webAcl: WebAclStack;
};

export class RemoteOutputStack extends cdk.Stack {
  public readonly webAclArn: string;

  public static makeOutputWebAclArn(webAcl: WebAclStack) {
    new cdk.CfnOutput(webAcl, outputKey, { value: webAcl.webAclArn });
  }

  constructor(scope: cdk.App, id: string, props: Props) {
    super(scope, id, props);

    this.addDependency(props.webAcl);
    const outputs = new RemoteOutputs(this, "Outputs", { stack: props.webAcl });
    const webAclArn = outputs.get(outputKey);

    this.webAclArn = webAclArn;
  }
}
