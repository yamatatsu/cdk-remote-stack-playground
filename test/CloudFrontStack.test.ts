import { Template } from "@aws-cdk/assertions";
import * as cdk from "@aws-cdk/core";
import { CloudFrontStack } from "../lib/CloudFront";

test("Snapshot Test", () => {
  const app = new cdk.App();
  // WHEN
  const stack = new CloudFrontStack(app, "WebAcl", { webAclArn: "test-arn" });

  // THEN
  expect(Template.fromStack(stack).toJSON()).toMatchSnapshot();
});
