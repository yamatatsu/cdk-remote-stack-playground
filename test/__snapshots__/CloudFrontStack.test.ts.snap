// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
Object {
  "Resources": Object {
    "DistributionCFDistribution882A7313": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "ConnectionAttempts": 3,
              "ConnectionTimeout": 10,
              "CustomOriginConfig": Object {
                "HTTPPort": 80,
                "HTTPSPort": 443,
                "OriginKeepaliveTimeout": 5,
                "OriginProtocolPolicy": "https-only",
                "OriginReadTimeout": 30,
                "OriginSSLProtocols": Array [
                  "TLSv1.2",
                ],
              },
              "DomainName": "dev.classmethod.jp",
              "Id": "origin1",
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "CloudFrontDefaultCertificate": true,
          },
          "WebACLId": "test-arn",
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
  },
}
`;
